### PR TITLE
perf(v2): convert synchronous filewrite to asynchronous

### PR DESCRIPTION
**Pull request by moonrailgun**
## Motivation

I encounter same problem like #2539 in alpha.50 and discover its have been resolved. But the solution to fix is sync way. 

### Have you read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/master/CONTRIBUTING.md#pull-requests)?

Yep

## Test Plan

Build website and make sure that the `build/blog/atom.xml`, `build/blog/rss.xml` and `build/sitemap.xml` files are still work correct.

## Related PRs

(If this PR adds or changes functionality, please take some time to update the docs at https://github.com/facebook/docusaurus, and link to your PR here.)

### CODE REVIEW

File: packages/docusaurus-plugin-content-blog/src/index.ts [hunk 1]
```diff
@@ -448,11 +448,9 @@ export default function pluginContentBlog(
             `${feedType}.xml`,
           );
           const feedContent = feedType === 'rss' ? feed.rss2() : feed.atom1();
-          try {
-            fs.writeFileSync(feedPath, feedContent);
-          } catch (err) {
+          return fs.outputFile(feedPath, feedContent).catch((err) => {
```
Comment (slorber) [inline packages/docusaurus-plugin-content-blog/src/index.ts:451]: Hi

What about using async/await here too. I think we should progressively update to async/await everywhere for consistency

Comment (moonrailgun) [reply packages/docusaurus-plugin-content-blog/src/index.ts:451]: Hi @slorber ,
I know about that. But this a `Promise.all`, and should return promise. I dont think `async/await` is suitable. 

`async/await` is good syntax but not fit any case

If you still persist in it, i will update it.

Thanks for you review

Comment (slorber) [reply packages/docusaurus-plugin-content-blog/src/index.ts:451]: What is the problem with Promise.all and async/await?

We already have this in other places already

Comment (slorber) [reply packages/docusaurus-plugin-content-blog/src/index.ts:451]: like in packages/docusaurus-plugin-content-docs/src/index.ts:464

Comment (moonrailgun) [reply packages/docusaurus-plugin-content-blog/src/index.ts:451]: Well, just think this too redundancy —— its unnecessary to write `async/await`.

But if that is a project rule, i will comply with it.

**Comment by slorber** [comment]
thanks

**Comment by moonrailgun** [comment]
my pleasure

**Pull request merged**